manticore/ethereum.py

```diff
@@ -655,17 +662,57 @@ def make_symbolic_value(self):
         return ABI.SValue
 
     @staticmethod
-    def compile(source_code, contract_name=None):
+    def compile(source_code, contract_name=None, libraries=None):
         ''' Get initialization bytecode from a Solidity source code '''
-        name, source_code, bytecode, runtime, srcmap, srcmap_runtime, hashes, abi, warnings = ManticoreEVM._compile(source_code, contract_name)
+        name, source_code, bytecode, runtime, srcmap, srcmap_runtime, hashes, abi, warnings = ManticoreEVM._compile(source_code, contract_name, libraries)
         return bytecode
 
     @staticmethod
-    def _compile(source_code, contract_name):
+    def _link(bytecode, libraries=None):
+        """
+        Fill bytecode placeholders with library addresses.
+
+        From https://solidity.readthedocs.io/en/v0.4.21/contracts.html:
+        `If the addresses are not given as arguments to the compiler, the 
+        compiled hex code will contain placeholders of the form __Set______
+        (where Set is the name of the library). The address can be filled
+        manually by replacing all those 40 symbols by the hex encoding of 
+        the address of the library contract.`
+        """
+        has_dependencies = '_' in bytecode
+        hex_contract = bytearray(bytecode, 'utf8')
+        if has_dependencies:
```

Comment:
nit style-wise can we maybe invert this to return early, so the main code here doesn't have to be indented?